<?xml version="1.0" encoding="utf-8"?>
<launch>
    <rosparam command="load" file="$(find task_switch)/config/xyztheta.yaml" />
    <group ns="theta1">
      <node pkg="task_switch" name="show_origin" type="central_origin_show.py" output="screen">
        <param name ="target/x" value="0"/>
        <param name ="target/y" value="0"/>
        <param name ="target/z" value="0"/>
        
        <param name="clock" value="20" />
      </node>
      <node pkg="task_switch" name="visualize_theta" type="pointcloud_origin_show.py" output="screen">
        <param name="center_x" value="0"/>
        <param name="center_y" value="2.5"/>
      </node>
    </group>

    <group ns="theta2">
      <node pkg="task_switch" name="show_origin" type="central_origin_show.py" output="screen">
        <param name ="target/x" value="-1.0"/>
        <param name ="target/y" value="1.0"/>
        <param name ="target/z" value="0"/>
        
        <param name="clock" value="20" />
      </node>
      <node pkg="task_switch" name="visualize_theta" type="pointcloud_origin_show.py" output="screen">
        <param name="center_x" value="-1.0"/>
        <param name="center_y" value="2.5"/>
      </node>
    </group>


      <param name="/agentNum" value="3"/>

      <param name="/x_min" value="-1.5"/>
      <param name="/x_max" value="1.5"/>
      <param name="/y_min" value="-1.5"/>
      <param name="/y_max" value="1.5"/>
      <param name="/mesh_acc/x" value="101"/>
      <param name="/mesh_acc/y" value="101"/>
  
        
      <include file="$(find task_switch)/launch/joy.launch"/>
  
      <node name="poseCollector" pkg="task_switch" type="posecollector.py" output="screen">
        <!-- for specify /bebop10*/virtualdrone/posestamped -->
        <!-- "preTopicName"bebop10*/"posestampedTopic" will be subscribed-->
        <param name="posestampedTopic" value="/virtualdrone/posestamped"/>
        <param name="preTopicName" value="/"/>
      </node>
  
  
      <node name="pcc_parameter" pkg="task_switch" type="task_switch_pcc_param_server" output="screen"/>
      <node name="charge_parameter" pkg="task_switch" type="task_switch_charge_param_server" output="screen"/>
      <node name="cbf_parameter" pkg="task_switch" type="task_switch_cbf_param_server" output="screen"/>
  
      <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen"/>
  
      <node pkg="task_switch" name="central" type="central.py" output="screen">
          <param name="clock" value="20"/>
      </node>
  
  
      
  
      <!-- Run simulation -->
      <include file="$(find task_switch)/launch/one.launch">
        <arg name="number" value="1"/>
        <arg name="initial_pose_x" value="1.0"/>
        <arg name="initial_pose_y" value="0.0"/>
      </include>
      
      <include file="$(find task_switch)/launch/one.launch">
        <arg name="number" value="2"/>
        <arg name="initial_pose_x" value="-1.0"/>
        <arg name="initial_pose_y" value="0.0"/>
      </include>
  
      <include file="$(find task_switch)/launch/one.launch">
        <arg name="number" value="3"/>
        <arg name="initial_pose_x" value="0.0"/>
        <arg name="initial_pose_y" value="0.5"/>
      </include>
  
  
      <!-- <node pkg="task_switch" name="visualize_surf" type="visualize_surf_python.py" output="screen"> -->
      <!--   <remap from="surf_value" to="/info"/> -->
      <!-- </node> -->
      <node pkg="task_switch" name="visualize_surf" type="visualize_surf_pointcloud.py" output="screen">
        <remap from="surf_value" to="/info"/>
      </node>
  
      <!-- <node pkg="task_switch" name="plotter" type="plotter.py" output="screen"> -->
      <!--   <remap from="surf_value" to="/info"/> -->
      <!-- </node> -->
  
      <node pkg="rviz" type="rviz" name="rviz" args="-d $(find task_switch)/config/sim_python.rviz"/>
</launch>
